### PR TITLE
Rename agent state property to lk.agent.state

### PR DESCRIPTION
**Pull request by bcherry**
Some back and forth here but we've landed on genericizing the state property and leaving the rest as-is.

### CODE REVIEW

File: packages/react/src/hooks/useAgent.ts [hunk 1]
```diff
@@ -11,61 +11,60 @@ import { useParticipantAttributes } from './useParticipantAttributes';
...
-export type VoiceAssistantState =
+export type AgentState =
   | 'disconnected'
   | 'connecting'
   | 'initializing'
   | 'listening'
   | 'thinking'
-  | 'speaking';
+  | 'speaking'
+  | string;
```
Comment (bcherry) [inline packages/react/src/hooks/useAgent.ts:21]: this change allows users to define their own agent states not included in our default enumeration and still take advantage of the `useAgent` hook

File: .changeset/cuddly-beers-cry.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"@livekit/components-react": patch
```
Comment (lukasIO) [inline .changeset/cuddly-beers-cry.md:2]: even if in beta, my hunch is we should mark this as `minor`

```suggestion
"@livekit/components-react": minor
```

File: packages/react/src/hooks/useAgent.ts (hunk 1 above)
Comment (bcherry) [reply packages/react/src/hooks/useAgent.ts]: @lukasIO I changed this back too to keep it simple, but I'm wondering:

what happens if the backend sends a value not in this list, but the frontend expects only one of these five types? I'm assuming nothing too bad but I'm not really sure. We will definitely add more types in the future, in one way or another.

**Review by lukasIO — APPROVED** [review]

File: packages/react/src/hooks/useAgent.ts (hunk 1 above)
Comment (lukasIO) [reply packages/react/src/hooks/useAgent.ts]: that should be fine, others will just be ignored

**Pull request merged**